### PR TITLE
classes: Add FileData object list to package class

### PR DESCRIPTION
**Pull request by PrajwalM2212**
This PR adds files property to package class. It adds add_file, remove_file, get_file_paths methods. Tests have been added for the changes made. The to_dict method has also been changed to account for the new files property

Closes https://github.com/vmware/tern/issues/559

Signed-off-by: PrajwalM2212 <prajwalmmath@gmail.com>

### CODE REVIEW

File: tern/classes/package.py [hunk 1]
```diff
@@ -90,11 +100,33 @@ def checksum(self):
...
         self.__checksum = checksum
 
+    def get_file_paths(self):
+        """Return a list of paths of all the files in a package"""
+        return list(map(lambda f: f.path, self.__files))
+
+    def add_file(self, file):
+        """Add a file to the list of files present in a package"""
+        if isinstance(file, FileData):
+            if file.path not in self.get_file_paths():
```
Comment (nishakm) [inline tern/classes/package.py]: I don't think I added this extra check in the `add_file` function in the `ImageLayer` class. Nice!

File: tern/classes/package.py [hunk 2]
```diff
@@ -90,11 +100,33 @@ def checksum(self):
     def checksum(self, checksum):
         self.__checksum = checksum
 
+    def get_file_paths(self):
+        """Return a list of paths of all the files in a package"""
+        return list(map(lambda f: f.path, self.__files))
```
Comment (nishakm) [inline tern/classes/package.py]: I think this is a bit overkill and hurts readability, particularly if you are casting the returned object multiple times. How about using list comprehension if you want a one-liner: `return [f.path for f in self.__files]`

**Comment by nishakm** [comment]
@PrajwalM2212 I asked for a small change in the code. Other than that, it looks good to me.

File: tern/classes/package.py (hunk 2 above)
Comment (PrajwalM2212) [reply tern/classes/package.py]: list comprehension is way better. Thanks

**Comment by PrajwalM2212** [comment]
@nishakm Changes made

**Pull request merged**